### ISSUE TITLE
Indented markdown code blocks in docstrings do not render correctly

### ISSUE THREAD

**Issue by gracepetryk** [microsoft/pyright#4718]
**Describe the bug**
When a docstring contains a triple backtick style code block ```, it only parses the code block correctly if the opening tag is indented to the same level as the opening quote of the docstring.

**To Reproduce**
Hover over the definition of the following function:
```python
def test():
    """
    A test function

    Example:

        ```python
        print(test())
        ```
    """
    pass
```
![image](https://user-images.githubusercontent.com/5047809/222282668-c2e824b9-62e6-48a6-82b7-9248f38a19a5.png)

**Expected behavior**
Code inside block gets syntax highlighting and other markdown styles.

This docstring renders as I would expect:
```python
def test():
    """
    A test function

    Example:

    ```python
        print(test())
    ```
    """
    pass
```
![image](https://user-images.githubusercontent.com/5047809/222283027-cc02bbb8-3046-4453-8752-f03c85f331e9.png)

**VS Code extension or command-line**
Both - neovim using pyright for it's language server fails to render docstrings formatted this way as well.

**Comment by erictraut** [comment]
@gracepetryk, thanks for the contribution.

This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published. It will also be included in a future release of pylance.
